DSA/Algorithms/Asymptotic_Analysis.md

```diff
@@ -0,0 +1,103 @@
+# Asymptotic Analysis of Algorithms
+
+
+### 🤔 What is Asymptotic Analysis?
+Asymptotic Analysis is the general methodolgy for studying the performance of algorithms, especially their run time and space requirements. It is used to determine an algorithm's **efficiency**, taking into consideration the best case, average case and the worst case inputs. 
+In asymptotic notations, we derive the complexity concerning the size of the input. In simple words, it helps us determine **how the running time (or space) would be affected as the input size goes on increasing**. </br>
+Remember: **Time complexity is NOT EQUAL TO running time of the algorithm !**
+
+It is important because: 
+- Takes into account **all possible inputs**.
+- Allows one to evaluate the efficiency of any algorithm in a way that is **independent of the hardware and software** environment.
```

Comment:
This Line is taken from https://pdfcoffee.com/data-structure-by-naveen-garg-pdf-free.html. Kindly change the speech lil bit , otherwise it's a plagirised content.

Reply:
Okay, made the necessary changes